src/WebJobs.Script/Utility.cs

```diff
@@ -626,6 +626,11 @@ public static bool IsSupportedRuntime(string workerRuntime, IEnumerable<RpcWorke
 
         private static bool ContainsFunctionWithWorkerRuntime(IEnumerable<FunctionMetadata> functions, string workerRuntime)
         {
+            // Codeless is allowed for all worker runtimes
+            if (functions.Any(f => string.Equals(f.Language, DotNetScriptTypes.Codeless, StringComparison.OrdinalIgnoreCase)))
```

Comment:
This doesn't seem right. This is called from Utility.IsSingleLanguage to ensure no mixed languages. If I have mixed Node/.NET and also a codeless, won't this short circuit the below required checks?

Reply:
This function checks if there's any function in the list that is acceptable for the current WorkerRuntime. So, say if you have `.NET` and `Node` functions, with `Node` as the worker rutime, it returns true as is. The `IsSingleLanguage` seems to check if it's a single language or if there is at least one function that matches the workerRuntime set.

In that sense, we are defining that `Codeless` is acceptable for all language workers. So, if you have `.NET`, `Node` and `Codeless` functions with worker runtime set to `Node`, that is allowed.
Additionally, if you have Worker Runtime set to `node` and you only have Codeless and .NET functions, that's fine too. Because Codeless is an acceptable function for Node worker runtime.

Reply:
I find the IsSingleLanguage check confusing. Only if the WorkerRuntime isn't set does it appear to verify that functions declare a single language. If it is set (the normal case), it returns true if there is at least one function for the specified language, but doesn't ensure there is a single, which doesn't match the behavior described by the method name. Your check seems to short circuit this - it'll always return true. @pragnagopa understands this logic best I believe since I think she wrote it. If she thinks this is correct, then OK :)

Reply:
If FUNCTIONS_WORKER_RUNTIME is set, it takes precedence and only need to check if there atleast one function matching the worker runtime set. If worker runtime is not set, the check enforces all the functions have same language.  Rename 'IsSingleLanguage` to 'IsSingleLanguageOrContainsWorkerRuntimeFunctions` ?

Do update tests to include cases for CodeLess functions

Reply:
Yes, good point. I will add tests. I can change the name to `IsSingleLanguageOrContainsWorkerRuntimeFunctions`, but is that a bit verbose?

I worry about tests that are `IsSingleLanguage_FunctionsList_Null_FunctionsWorkerRuntime_Throws_ArgumentNullException` will become `IsSingleLanguageOrContainsWorkerRuntimeFunctions_FunctionsList_Null_FunctionsWorkerRuntime_Throws_ArgumentNullException`, :P unless I don't rename them.

Could we reduce the scope of this function to only check for if there's a single function language? Then let the caller call the other `ContainsFunctionWithWorkerRuntime` if needed?
Looks like only one caller `Utility.VerifyFunctionsMatchSpecifiedLanguage` actually passes a `workerRuntime`, and that can manually call `ContainsFunctionWithWorkerRuntime`. The other caller just sets `workerRuntime` to `null` and would never call `ContainsFunctionWithWorkerRuntime` anyway.